src/parser.ts

```diff
@@ -0,0 +1,353 @@
+    private controlPointTransition: { [key: string]: Curve };
+    private controlPointTriggerable: { [key: string]: Curve };
+    private stateMachines: { [key: string]: StateMachine };
+    private states: { [key: string]: State };
+    private transitionPatternStates: { [key: string]: State };
+    private linksLabel: Array<LinkLabelTemplate>;
+    private componentGraphicalModel: ComponentGraphicalModel;
+
+    private componentName: string;
+    private finalStates: Array<String>;
+    private entryPointState: string;
+    private entryPointStateMachine: string;
+    private stateMachineNames: Array<string>;
+    private linkDataArray: Array<LinkDataArrayTemplate>;
+    private nodeDataArray: Array<NodeDataArrayTemplate>;
+
+    constructor(componentGraphicalModel: ComponentGraphicalModel) {
+        this.componentGraphicalModel = componentGraphicalModel;
+    }
+
+    parse() {
+        this.parseGraphical();
+        this.parseModel();
+    }
+
+    private parseGraphical(): void {
+        this.locations = {};
+        const xmlGraphicalDom = (new DOMParser()).parseFromString(this.componentGraphicalModel.graphical, "text/xml");
+        let stateGraphicalData = xmlGraphicalDom.getElementsByTagName(graphicalTags.States)[0].getElementsByTagName(graphicalTags.StateGraphicalData);
+        for (let i = 0; i < stateGraphicalData.length; i++) {
+            const id = stateGraphicalData[i].getAttribute(graphicalTags.Id);
+            this.locations[id] = {
+                x: parseFloat(stateGraphicalData[i].getAttribute(graphicalTags.CenterX)),
+                y: parseFloat(stateGraphicalData[i].getAttribute(graphicalTags.CenterY))
+            };
+        }
+
+        stateGraphicalData = xmlGraphicalDom.getElementsByTagName(graphicalTags.TransitionPatternStates)[0].getElementsByTagName(graphicalTags.StateGraphicalData);
+        for (let i = 0; i < stateGraphicalData.length; i++) {
+            const id = stateGraphicalData[i].getAttribute(graphicalTags.Id);
+            this.locations[id] = {
+                x: parseFloat(stateGraphicalData[i].getAttribute(graphicalTags.CenterX)),
+                y: parseFloat(stateGraphicalData[i].getAttribute(graphicalTags.CenterY))
+            };
+        }
+
+        let transitionGraphicalData;
+
+        transitionGraphicalData = xmlGraphicalDom.getElementsByTagName(graphicalTags.Links)[0]
+            .getElementsByTagName(graphicalTags.TransitionGraphicalData);
+        this.controlPointTransition = this.getControlPointTransition(transitionGraphicalData);
+
+        transitionGraphicalData = xmlGraphicalDom.getElementsByTagName(graphicalTags.TransversalLinks)[0]
+            .getElementsByTagName(graphicalTags.TransitionGraphicalData);
+        this.controlPointTriggerable = this.getControlPointTransition(transitionGraphicalData);
+    }
+
+    private parseModel(): void {
+        const scxmlDom = (new DOMParser()).parseFromString(this.componentGraphicalModel.model, "text/xml");
```

Comment:
Why using DOMParser ? and not using xml2js like our js api.
It's easier to maintain (instead of doing manual parsing)